recipes/bison/all/conanfile.py

```diff
@@ -0,0 +1,81 @@
+
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+import glob
+import os
+import shutil
+
+
+class ConanFileDefault(ConanFile):
+    name = "bison"
+    url = "https://github.com/conan-io/conan-center-index   "
+    homepage = "https://www.gnu.org/software/bison/"
+    description = "Bison is a general-purpose parser generator"
+    topics = ("conan", "bison", "parser")
+    license = "GPL-3.0-or-later"
+    exports_sources = ["patches/*.patch"]
+    _source_subfolder = "source_subfolder"
+    requires = ("m4/1.4.18", "flex/2.6.4")
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"fPIC": [True, False]}
+    default_options = {"fPIC": True}
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "bison-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def build(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+        self._build_configure()
+
+    def _build_configure(self):
+        true = tools.which("true") or "/bin/true"
+        args = ["HELP2MAN=%s" % true, "MAKEINFO=%s" % true, "--disable-nls"]
+        build = None
+        host = None
+
+        env_build = AutoToolsBuildEnvironment(self)
+        with tools.chdir(self._source_subfolder):
+            tools.replace_in_file("Makefile.in",
+                                  "dist_man_MANS = $(top_srcdir)/doc/bison.1",
+                                  "dist_man_MANS =")
+            tools.replace_in_file(os.path.join("src", "yacc.in"),
+                                  "@prefix@",
+                                  "${}_ROOT".format(self.name.upper()))
+            tools.replace_in_file(os.path.join("src", "yacc.in"),
+                                  "@bindir@",
+                                  "${}_ROOT/bin".format(self.name.upper()))
+
+            env_build.configure(args=args, build=build, host=host)
+            env_build.make()
+            env_build.install()
+
+    def package(self):
+        self.copy(pattern="COPYING", dst="licenses", src=self._source_subfolder)
+        
+        #tools.rmdir(os.path.join(self.package_folder, "share"))
+        shutil.move(os.path.join(self.package_folder, "share"), os.path.join(self.package_folder, "bin", "share"))
+
+
+    def package_info(self):
+        self.cpp_info.libs = ["y"]
+
+        self.output.info('Setting BISON_ROOT environment variable: {}'.format(self.package_folder))
+        self.env_info.BISON_ROOT = self.package_folder
+
+        bindir = os.path.join(self.package_folder, "bin")
+        self.output.info('Appending PATH environment variable: {}'.format(bindir))
+        self.env_info.PATH.append(bindir)
+
```

Comment:
```suggestion
```
For cross building using conan, this is not needed anymore.
See https://github.com/conan-io/conan-center-index/pull/1179#issuecomment-603848593 and https://github.com/conan-io/conan-center-index/issues/1136#issuecomment-603897062

Reply:
But this is not ready yet. We need support for current Conan release versions too

Reply:
This will recipe sets `os`, `arch`, `compiler`, `build_type`, not `os_build`, `arch_build`.

Reply:
Maybe I'm missing the point. 

You suggested to remove code which adds the executable to the PATH. Conan is not adding the bin directories to the PATH automatically, so we still need to do it manually.

If you compile the executables for the same machine or if you cross-compile does not change this situation

🤔

Reply:
IIUC, the current situation regarding cross compilation is: we do not create separate recipes for installer and libraries. As a consequence we always do only one build for both the build platform and the host platform, so I don't understand how os_build and arch_build would be used. Could someone please clarify this for me ?

Reply:
@Croydon

> You suggested to remove code which adds the executable to the PATH. Conan is not adding the bin directories to the PATH automatically, so we still need to do it manually.

Indeed, removing these vars should only be done when the feature proposed in https://github.com/conan-io/conan-center-index/pull/1179#issuecomment-603848593 lands.

@ericLemanissier 
> IIUC, the current situation regarding cross compilation is: we do not create separate recipes for installer and libraries. As a consequence we always do only one build for both the build platform and the host platform, so I don't understand how os_build and arch_build would be used. Could someone please clarify this for me ?

By mentioning `arch_build`/`os_build`, I just wanted to say that the proposed recipe here will not work for cross building from e.g. x86_64 to arm64, using current conan. (Future conan will allow it). Imho, this recipe is fine.